### PR TITLE
Adding Stylelint

### PR DESCRIPTION
**Pull request by kaytcampbell**
:art: 🐘 
Adds the stylelint linter to travis, also linted all our scss. You can read up on each of the rules here: https://stylelint.io/user-guide/rules/

### PR Review Checklist
To be completed by the person who opens the PR. Use strikethroughs for items that are not applicable in list.
- [x] I have reviewed my own PR to check syntax & logic, removed unnecessary/old code, made sure everything aligns with our style guide and BEM.
- [ ] ~I've added new components to the style guide (or have a PR in to add them).~
- [x] Any applicable version numbers have been updated.
- [x] I've tested on mobile, tablet, and desktop, as well as across all of the browsers we support.
- [ ] ~I've proof-read all text for legibility, proper grammar, punctuation, and capitalization (titlecase).~
- [x] I have written a detailed PR message explaining what is being changed and why
- [ ] ~I’ve linked to any relevant VSTS tickets~
- [x] I’ve added the appropriate review symbols to my PR - (elephant) for dev review, (art) for design

### CODE REVIEW

**Comment by sineade** [comment]
:book:

**Comment by jessedoyle** [comment]
:book:

**Comment by jessedoyle** [comment]
:elephant::elephant::elephant:  - Also, we can remove all vendor prefixes as we use https://github.com/ai/autoprefixer-rails to automatically add them before deployment!

File: app/assets/stylesheets/layout_components/comparison-radios.scss [hunk 1]
```diff
@@ -213,59 +213,59 @@
...
     display: inline-block;
     width: 85%;
     margin: 9px auto;
   }
 
-  &__input:checked{
-    &~label{
-      .comparison-radios__panel{
+  &__input:checked {
+    & ~ label {
```
Comment (DaveAMA) [inline app/assets/stylesheets/layout_components/comparison-radios.scss]: I just looked up what `~` does, that is great!

File: app/assets/stylesheets/layout_components/features.scss [hunk 2]
```diff
@@ -27,77 +27,76 @@
...
+  &--gift-cards {
     background-image: asset-url(gatekeeper/gift-card);
   }
 }
 
-@media (-webkit-min-device-pixel-ratio: 2),
-(min-resolution: 192dpi){
-  #feature__image{
-    &--devices{
+@media (-webkit-min-device-pixel-ratio: 2), (min-resolution: 192dpi) {
```
Comment (DaveAMA) [inline app/assets/stylesheets/layout_components/features.scss]: So is the idea here that multiple selectors / classes which are separated by commas should be on new lines, but media query attributes separated by commas are on the same line?

File: app/assets/stylesheets/layout_components/forms.scss [hunk 3]
```diff
@@ -159,8 +155,8 @@ input[type="checkbox"] + label,
   cursor: pointer;
   padding: 0 $base-padding/2 0 $base-padding/2;
 
-  &:before{
-    content: ' ';
+  &::before {
+    content: " ";
```
Comment (DaveAMA) [inline app/assets/stylesheets/layout_components/forms.scss]: really minor, but this is inconsistent with `content: " ";` below on line 212

File: app/assets/stylesheets/layout_components/forms.scss [hunk 4]
```diff
@@ -188,38 +184,38 @@ input[type="checkbox"] + label,
...
-input[type="radio"]:checked + label{
+input[type="radio"]:checked + label {
   color: $stormcloud;
   background-color: $sky;
   border: 2px solid darken($sky, 10%);
 
-  &:before{
-    content: '';
+  &::before {
+    content: "";
```
Comment (DaveAMA) [inline app/assets/stylesheets/layout_components/forms.scss]: see above comment on line 159

File: app/assets/stylesheets/layout_components/forms.scss [hunk 5]
```diff
@@ -366,98 +361,95 @@ input[type="checkbox"]{
...
+.minute {
   float: right;
 }
 
-.datetime{
-  &.month, &.day, &.year{
+.datetime {
+  &.month,
+  &.day,
+  &.year {
```
Comment (DaveAMA) [inline app/assets/stylesheets/layout_components/forms.scss]: 👍 Nice

File: app/assets/stylesheets/layout_components/forms.scss [hunk 6]
```diff
@@ -366,98 +361,95 @@ input[type="checkbox"]{
...
 textarea,
-select{
+select {
   margin-bottom: 0;
   -webkit-appearance: none;
   -moz-appearance: none;
   appearance: none;
 }
 
-select.has-placeholder:nth-child(1){
```
Comment (DaveAMA) [inline app/assets/stylesheets/layout_components/forms.scss]: Why was this removed?

**Comment by DaveAMA** [comment]
📖

File: app/assets/stylesheets/layout_components/notices.scss [hunk 7]
```diff
@@ -9,104 +9,151 @@
...
+    opacity: 1;
+  }
+
+  90% {
+    opacity: 1;
+  }
+
+  100% {
+    opacity: 0;
+  }
```
Comment (DaveAMA) [inline app/assets/stylesheets/layout_components/notices.scss]: So big!

File: app/assets/stylesheets/layout_components/notification.scss [hunk 8]
```diff
@@ -1,43 +1,33 @@
...
-
-.notification{
-  &__badge{
-    @include badge;
-    background: $brand-red;
-    color: $white;
-    line-height: 1.5;
-    font-weight: bolder;
-  }
-}
```
Comment (DaveAMA) [inline app/assets/stylesheets/layout_components/notification.scss]: 👍

**Comment by DaveAMA** [comment]
💩

File: app/assets/stylesheets/layout_components/features.scss (hunk 2 above)
Comment (kaytcampbell) [reply app/assets/stylesheets/layout_components/features.scss]: yeah, my thought was that here they both relate back to the @media so it made sense, whereas usually if we have different classes separated by commas they aren't (necessarily) related to each other

File: app/assets/stylesheets/layout_components/forms.scss (hunk 3 above)
Comment (kaytcampbell) [reply app/assets/stylesheets/layout_components/forms.scss]: there is actually a subtle difference here, in this instance it is saying put a space before the content, whereas the one below is saying don't put a space, so they don't need to match

File: app/assets/stylesheets/layout_components/forms.scss (hunk 6 above)
Comment (kaytcampbell) [reply app/assets/stylesheets/layout_components/forms.scss]: oh, because I realized this was something that I added a long time ago but it doesn't work as expected/we don't use it anywhere

File: app/assets/stylesheets/layout_components/forms.scss (hunk 3 above)
Comment (DaveAMA) [reply app/assets/stylesheets/layout_components/forms.scss]: Ah, makes sense.

**Comment by sineade** [comment]
:art: :art: :art:
